src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt

```diff
@@ -7,18 +7,17 @@ import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.core.ParameterizedTypeReference
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
+import uk.gov.justice.digital.hmpps.hmppstier.domain.enums.ComplexityFactor
+import uk.gov.justice.digital.hmpps.hmppstier.domain.enums.ComplexityFactor.IOM_NOMINAL
 import java.math.BigDecimal
 import java.time.LocalDate
 
 @Component
 class CommunityApiClient(@Qualifier("communityWebClientAppScope") private val webClient: WebClient) {
 
-  fun getRegistrations(crn: String): Collection<Registration> {
-    return getRegistrationsCall(crn).sortedByDescending { it.startDate }
-      .also {
-        log.info("Fetched ${it.size} Registrations for $crn")
-      }
-  }
+  fun getRegistrations(crn: String): Pair<List<Registration>, List<Registration>> =
+    getRegistrationsCall(crn).sortedByDescending { it.startDate }
+      .partition { ComplexityFactor.from(it.type.code) == IOM_NOMINAL }
```

Comment:
Doing this once rather than twice

Reply:
I can see the benefit but I think this worsens coupling and cohesion, having a client that knows about change and protect